src/numerics/include/metaphysicl/dualnumber_surrogate.h

```diff
@@ -57,17 +57,33 @@ inline DualNumberSurrogate<T, D>::DualNumberSurrogate(DualNumber<T2, D2> && dn,
 }
 
 template <typename T, typename D>
-inline DualNumberSurrogate<T, D>::DualNumberSurrogate(DualNumberSurrogate<T, D> & dns)
-  : _value(dns.value()), _derivatives(dns.derivatives())
+inline DualNumberSurrogate<T, D>::DualNumberSurrogate(const DualNumberSurrogate<T, D> & dns)
+  : _value(dns._value), _derivatives(dns._derivatives)
```

Comment:
Why these changes?

Reply:
Replacing accessors with raw members, specifically.  Do we not have const versions of the accessors?